### PR TITLE
Simplify automated release workflow

### PR DESCRIPTION
**Pull request by xin-hedera**
**Detailed description**:

- Use the newly added `pre-release` output in action `terradatum/semver-action`
- Only create a PR to bump master snapshot version when the first time the release branch is created
- Remove `Update Changelog` step

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: .github/workflows/auto-release.yml [hunk 1]
```diff
@@ -139,34 +117,6 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           MILESTONE_NUMBER: ${{ steps.milestone.outputs.number }}
 
-      - name: Update Changelog
```
Comment (steven-sheehy) [inline .github/workflows/auto-release.yml:142]: `env.CHANGELOG` can also be deleted

File: .github/workflows/auto-release.yml [hunk 2]
```diff
@@ -180,11 +130,12 @@ jobs:
       - name: Create Github Release
         uses: ncipollo/release-action@v1
         with:
-          bodyFile: ${{ env.GITHUB_RELEASE_NOTES }}
+          bodyFile: ${{ env.RELEASE_NOTES_FULLNAME }}
```
Comment (steven-sheehy) [inline .github/workflows/auto-release.yml:133]: Can drop `env.RELEASE_NOTES_FULLNAME` if you change to below:

```suggestion
          bodyFile: ${{ env.RELEASE_NOTES_FILENAME }}.md
```

File: .github/workflows/auto-release.yml [hunk 3]
```diff
@@ -139,34 +117,6 @@ jobs:
...
-          $RELEASE_NOTES
-
-          $CHANGELOG_PREVIOUS
-          EOF
-
-          echo "GITHUB_RELEASE_NOTES=$GITHUB_RELEASE_NOTES" >> $GITHUB_ENV
-
       - name: Commit and Tag
         uses: stefanzweifel/git-auto-commit-action@v4
         with:
```
Comment (steven-sheehy) [inline .github/workflows/auto-release.yml:122]: Have we made any attempt to submit the changes in `xin-hedera/release-notes-generator-action` upstream?

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: .github/workflows/auto-release.yml (hunk 1 above)
Comment (xin-hedera) [reply .github/workflows/auto-release.yml:142]: deleted

File: .github/workflows/auto-release.yml (hunk 2 above)
Comment (xin-hedera) [reply .github/workflows/auto-release.yml]: updated

File: .github/workflows/auto-release.yml (hunk 3 above)
Comment (xin-hedera) [reply .github/workflows/auto-release.yml:120]: not yet. I'll create a PR.

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by ijungmann — APPROVED** [review]

**Review by Nana-EC — APPROVED** [review]
LGTM

**Pull request merged**